From 56c6873507918848f769f129fcc4fd7c572bde07 Mon Sep 17 00:00:00 2001
From: Jakob Gruber <jgruber@chromium.org>
Date: Mon, 28 Feb 2022 11:09:39 +0100
Subject: [PATCH] Skip src/maglev for v8_check_header_includes

Maglev is currently enabled only on x64; BUILD.gn knows about this
through v8_enable_maglev, but v8_check_header_includes doesn't. It
thus tries to compile maglev files on platforms that don't have maglev
support yet.

Add an explicit maglev exclude rule until we support other platforms.

Bug: v8:7700
Change-Id: Iac991741c7d630dc4ed9f9fbf6df7656853cc743
Reviewed-on: https://chromium-review.googlesource.com/c/v8/v8/+/3494536
Commit-Queue: Jakob Gruber <jgruber@chromium.org>
Auto-Submit: Jakob Gruber <jgruber@chromium.org>
Reviewed-by: Leszek Swirski <leszeks@chromium.org>
Commit-Queue: Leszek Swirski <leszeks@chromium.org>
Cr-Commit-Position: refs/heads/main@{#79305}
---
 tools/generate-header-include-checks.py | 17 ++++++++++-------
 1 file changed, 10 insertions(+), 7 deletions(-)

diff --git a/tools/generate-header-include-checks.py b/tools/generate-header-include-checks.py
index 42c118c9d5e..2d6f218e116 100755
--- a/tools/generate-header-include-checks.py
+++ b/tools/generate-header-include-checks.py
@@ -37,14 +37,17 @@ AUTO_EXCLUDE = [
   'src/trap-handler/trap-handler-simulator.h',
 ]
 AUTO_EXCLUDE_PATTERNS = [
-  'src/base/atomicops_internals_.*',
-  # TODO(petermarshall): Enable once Perfetto is built by default.
-  'src/libplatform/tracing/perfetto*',
+    'src/base/atomicops_internals_.*',
+    # TODO(petermarshall): Enable once Perfetto is built by default.
+    'src/libplatform/tracing/perfetto*',
+    # TODO(v8:7700): Enable once Maglev is built by default.
+    'src/maglev/.*',
 ] + [
-  # platform-specific headers
-  '\\b{}\\b'.format(p) for p in
-    ('win', 'win32', 'ia32', 'x64', 'arm', 'arm64', 'mips', 'mips64', 's390',
-     'ppc', 'riscv64', 'loong64')]
+    # platform-specific headers
+    '\\b{}\\b'.format(p)
+    for p in ('win', 'win32', 'ia32', 'x64', 'arm', 'arm64', 'mips', 'mips64',
+              's390', 'ppc', 'riscv64', 'loong64')
+]
 
 args = None
 def parse_args():
-- 
2.35.1

